### PR TITLE
feat: Add azure_resources_resource_groups table

### PR DESCRIPTION
**Pull request by hermanschaaf**
This adds a new `azure_resources_resource_groups` table alongside the existing `azure_resources_resources` table.

Closes https://github.com/cloudquery/cloudquery/issues/9142

### CODE REVIEW

**Review by erezrokah — COMMENTED** [review]
Thanks for the PR @hermanschaaf, we already list the resource groups during initialization (since we have a resource that multiplexes over those), see https://github.com/cloudquery/cloudquery/blob/49d647730a85ea6fae51e97194ba61c0625d1331/plugins/source/azure/client/client.go#L76

At the moment we only keep the names, but we could keep the whole object and return it in the table resolver instead of doing the listing again, WDYT?

See example in https://github.com/cloudquery/cloudquery/blob/49d647730a85ea6fae51e97194ba61c0625d1331/plugins/source/azure/resources/services/subscription/subscriptions_fetch.go#L14

**Review by erezrokah — APPROVED** [review]

File: plugins/source/azure/client/helpers.go [hunk 1]
```diff
@@ -8,7 +8,7 @@ import (
 // this is used to hook ParseResourceGroup and to have easier codegen
 var debug = false
 
-const resourceIDPatternText = `(?i)subscriptions/(.+)/resourceGroups/(.+)/providers/(.+?)/(.+?)/(.+)`
+const resourceIDPatternText = `(?i)subscriptions/(.+)/ResourceGroups/(.+)/providers/(.+?)/(.+?)/(.+)`
```
Comment (disq) [inline plugins/source/azure/client/helpers.go:11]: Why this change? was it broken before? Tests set `debug` so this code doesn't really run without a real fetch.

Also we could remove the `const` and just have the string inside `regexp.MustCompile` below, it's not used anywhere else anyway.

Comment (hermanschaaf) [reply plugins/source/azure/client/helpers.go]: Reverted
